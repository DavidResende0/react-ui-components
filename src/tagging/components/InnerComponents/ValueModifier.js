import React from 'react';
import PropTypes from 'prop-types';
import { Col, ControlLabel, FormGroup } from 'patternfly-react';
import ValueSelector from './ValueSelector';
import { __ } from '../../../global-functions';
import TaggingPropTypes from '../TaggingPropTypes';

const ValueModifier = ({
  values,
  onTagValueChange,
  selectedTagValue,
  multiValue,
  valueLabel,
}) => (
  <FormGroup>
    <Col xs={12} md={4} lg={4}>
      <ControlLabel>{valueLabel}</ControlLabel>
    </Col>
    <Col xs={12} md={8} lg={8}>
      <ValueSelector
        values={values}
        onTagValueChange={onTagValueChange}
        selectedOption={selectedTagValue}
        multiValue={multiValue}
      />
    </Col>
  </FormGroup>
);

ValueModifier.propTypes = {
  selectedTagValue: TaggingPropTypes.value,
  onTagValueChange: PropTypes.func.isRequired,
  valueLabel: PropTypes.string,
  multiValue: PropTypes.bool,
  values: PropTypes.arrayOf(TaggingPropTypes.value).isRequired,
};

ValueModifier.defaultProps = {
  valueLabel: __('Value'),
  multiValue: true,
};

export default ValueModifier;
