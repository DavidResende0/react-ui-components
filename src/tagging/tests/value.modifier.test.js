import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ValueModifier from '../components/valueModifier';

const selectedTagValue = { description: 'Duck', id: 1 };
const onTagValueChange = jest.fn();
const tagValues = [
  { description: 'Duck', id: 1 },
  { description: 'Cat', id: 2 },
  { description: 'Dog', id: 3 },
];

describe('TagCategory Component', () => {
  it('match snapshot', () => {
    const tree = mount(<ValueModifier
      onTagValueChange={onTagValueChange}
      selectedTagValue={selectedTagValue}
      multiValue={false}
      tagValues={tagValues}
    />);

    expect(toJson(tree)).toMatchSnapshot();
  });
});
