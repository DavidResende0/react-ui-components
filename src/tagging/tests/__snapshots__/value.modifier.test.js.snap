// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCategory Component match snapshot 1`] = `
<ValueModifier
  multiValue={false}
  onTagValueChange={[MockFunction]}
  selectedTagValue={
    Object {
      "description": "Duck",
      "id": 1,
    }
  }
  tagValues={
    Array [
      Object {
        "description": "Duck",
        "id": 1,
      },
      Object {
        "description": "Cat",
        "id": 2,
      },
      Object {
        "description": "Dog",
        "id": 3,
      },
    ]
  }
  valueLabel="Value"
>
  <FormGroup
    bsClass="form-group"
  >
    <div
      className="form-group"
    >
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={4}
        xs={12}
      >
        <div
          className="col-lg-6 col-md-4 col-xs-12"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            <label
              className="control-label"
            >
              Value
            </label>
          </ControlLabel>
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        lg={6}
        md={8}
        xs={12}
      >
        <div
          className="col-lg-6 col-md-8 col-xs-12"
        >
          <ValueSelector
            multiValue={false}
            onTagValueChange={[MockFunction]}
            selectedOption={
              Object {
                "description": "Duck",
                "id": 1,
              }
            }
            tagValues={
              Array [
                Object {
                  "description": "Duck",
                  "id": 1,
                },
                Object {
                  "description": "Cat",
                  "id": 2,
                },
                Object {
                  "description": "Dog",
                  "id": 3,
                },
              ]
            }
          >
            <Select
              arrowRenderer={[Function]}
              autosize={true}
              backspaceRemoves={true}
              backspaceToRemoveMessage="Press backspace to remove {label}"
              clearAllText="Clear all"
              clearRenderer={[Function]}
              clearValueText="Clear value"
              clearable={false}
              closeOnSelect={true}
              deleteRemoves={true}
              delimiter=","
              disabled={false}
              escapeClearsValue={true}
              filterOptions={[Function]}
              ignoreAccents={true}
              ignoreCase={true}
              inputProps={Object {}}
              isLoading={false}
              joinValues={false}
              label="Duck"
              labelKey="label"
              matchPos="any"
              matchProp="any"
              menuBuffer={0}
              menuRenderer={[Function]}
              multi={false}
              name="form-field-name"
              noResultsText="No results found"
              onBlurResetsInput={true}
              onChange={[Function]}
              onCloseResetsInput={true}
              onSelectResetsInput={true}
              openOnClick={true}
              optionComponent={[Function]}
              options={
                Array [
                  Object {
                    "label": "Duck",
                    "value": 1,
                  },
                  Object {
                    "label": "Cat",
                    "value": 2,
                  },
                  Object {
                    "label": "Dog",
                    "value": 3,
                  },
                ]
              }
              pageSize={5}
              placeholder="Select..."
              removeSelected={true}
              required={false}
              rtl={false}
              scrollMenuIntoView={true}
              searchable={true}
              simpleValue={false}
              tabSelectsValue={true}
              trimFilter={true}
              value={1}
              valueComponent={[Function]}
              valueKey="value"
            >
              <div
                className="Select has-value is-searchable Select--single"
              >
                <input
                  disabled={false}
                  key="hidden.0"
                  name="form-field-name"
                  type="hidden"
                  value="1"
                />
                <div
                  className="Select-control"
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                >
                  <span
                    className="Select-multi-value-wrapper"
                    id="react-select-2--value"
                  >
                    <Value
                      disabled={false}
                      id="react-select-2--value-item"
                      instancePrefix="react-select-2-"
                      onClick={null}
                      placeholder="Select..."
                      value={
                        Object {
                          "label": "Duck",
                          "value": 1,
                        }
                      }
                    >
                      <div
                        className="Select-value"
                      >
                        <span
                          aria-selected="true"
                          className="Select-value-label"
                          id="react-select-2--value-item"
                          role="option"
                        >
                          Duck
                        </span>
                      </div>
                    </Value>
                    <AutosizeInput
                      aria-activedescendant="react-select-2--value"
                      aria-expanded="false"
                      aria-haspopup="false"
                      aria-owns=""
                      className="Select-input"
                      injectStyles={true}
                      minWidth="5"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      role="combobox"
                      value=""
                    >
                      <div
                        className="Select-input"
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-activedescendant="react-select-2--value"
                          aria-expanded="false"
                          aria-haspopup="false"
                          aria-owns=""
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          required={false}
                          role="combobox"
                          style={
                            Object {
                              "boxSizing": "content-box",
                              "width": "5px",
                            }
                          }
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        />
                      </div>
                    </AutosizeInput>
                  </span>
                  <span
                    className="Select-arrow-zone"
                    onMouseDown={[Function]}
                  >
                    <span
                      className="Select-arrow"
                      onMouseDown={[Function]}
                    />
                  </span>
                </div>
              </div>
            </Select>
          </ValueSelector>
        </div>
      </Col>
    </div>
  </FormGroup>
</ValueModifier>
`;
